src/chain/chaininterface.rs

```diff
@@ -21,13 +22,18 @@ pub trait ChainWatchInterface: Sync + Send {
 	/// Indicates that a listener needs to see all transactions.
 	fn watch_all_txn(&self);
 
-	/// Sends a transaction out to (hopefully) be mined
-	fn broadcast_transaction(&self, tx: &Transaction);
-
 	fn register_listener(&self, listener: Weak<ChainListener>);
 	//TODO: unregister
 }
 
+/// An interface to send a transaction to connected Bitcoin peers.
+/// This is for final settlement. An error might indicate that no peers can be reached or
+/// that peers rejected the transaction.
+pub trait BroadcasterInterface: Sync + Send {
+	/// Sends a transaction out to (hopefully) be mined
+	fn broadcast_transaction(&self, tx: &Transaction) -> Result<(), Box<Error>>;
```

Comment:
I'm not sure if its useful to return a Result here - Bitcoin transaction broadcasting is always somewhat a "fire and see if anything happens" process. In other words, there is always a high chance of undetectable failure, so trying to return failure is a bit useless, or to put it another way, is there any useful handling you can do except to try again later (which should happen anyway?). Maybe I'm just being idealistic about it?

Reply:
There are immediate error conditions that would better be communicated back, such as no connection to any peer or all peers rejected the message. An Ok result surely does not mean any confirmation, but without it does not even  make sense to wait if anything happens as it won't.